### PR TITLE
Add Firefox support (Fixes #22)

### LINKED ISSUE #22: Publish to Firefox?
Considering the simplicity of this extension I would assume that it would take little to no work to get this working on firefox. Could you please publish this on the Firefox store?

### PR DESCRIPTION
**Pull request by aaronraimist**
This PR adds support for Firefox, fixing issue #22. The add-on works fine with the same code in Firefox, just a little bit needs to be added to the `manifest.json` file. I assume it will still build and behave correctly in Chrome after this PR but I have not tested that.

I went ahead and [published the add-on to AMO (the Firefox add-on store)](https://addons.mozilla.org/en-US/firefox/addon/don-t-fuck-with-paste/). I am happy to serve as the maintainer of the Firefox version. Jacob, if you would like to be added as a maintainer on AMO, you can send me your email address. I think I gave you proper credit in the description but if you want something changed let me know.

The add-on will be available to download on AMO once it has been reviewed by Mozilla. At the time of this PR the add-on is at queue position 460 of 460.

I need to find a little graphic to put in the README next to the "available in the chrome web store" one. If anyone knows of a good one for Firefox add-ons, let me know.

### CODE REVIEW

**Comment by jswanner** [comment]
Hey @aaronraimist, thanks for the PR.

I'm currently in the process of a big change to this extension, and unfortunately the new version doesn't seem to work on Firefox. I'll try spending some more time to get it working on Firefox, but if I'm not successful, I'd rather not add support for a browser that would be limiting what I can do with the extension.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# Windows Installer files
+*.cab
+*.msi
+*.msm
+*.msp
+
+# Windows shortcuts
+*.lnk
+
+# End of https://www.gitignore.io/api/osx,linux,windows
```
Comment (jswanner) [inline .gitignore]: None of the stuff you are ignoring in this file is relevant to this project. An ignore file like this is better served as your personal global git ignore: https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

In repositories themselves, I prefer to only ignore things that makes sense for that project: intermediate files that are generated, logs, etc.

**Comment by singron** [comment]
What's the status here? AFAIK the big change @jswanner mentioned was version 2, and this PR has since incorporated that. The only actionable feedback on this PR has been the comment on the huge gitignore. @aaronraimist if you really don't want to remove that gitignore, could you open a new PR with it so that discussion doesn't block this one?

**Comment by aaronraimist** [comment]
Well sure I'm happy to close the PR but I think @jswanner has been pretty clear that he is only going to maintain a chrome version of the extension. I've been maintaining a Firefox version for several months now which you are welcome to use so I think you can also just go ahead and close #22.

**Comment by rugk** [comment]
> @jswanner has been pretty clear that he is only going to maintain a chrome version of the extension

I think this was not so clear.
And having [a Firefox fork](https://github.com/jswanner/DontFuckWithPaste) now is a little sad as I guess it's 99% compatible.

With [this polyfill, if needed](https://github.com/mozilla/webextension-polyfill/) you can patch the Chrome API to also behave like Promises.
But if you did not use it in the first place, it should also work as such. Firefox WebExtension API is very compatible to Chrome's one.

So what exactly was a blocker here to integrate it upstream, @jswanner? I'd like to unify the communities here and ease the maintenance task.

So itt [should be quite easy to port](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/Porting_a_Google_Chrome_extension) your extension to Firefox, as [there are not that many differences](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/Chrome_incompatibilities). Developing [cross-browser extensions is not that hard](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Build_a_cross_browser_extension), actually.

**Pull request closed without merging**